src/include/security/dialogs.rb

```diff
@@ -514,10 +515,12 @@ def OverviewDialog
               Builtins.y2milestone("Client returned %1", client_ret)
 
               if client_ret == :next || client_ret == :ok ||
-                  client_ret == :finish
+                  client_ret == :finish || client_ret == true
                 # update the current value
-                if Builtins.haskey(@link_update_mapping, ret)
-                  Builtins.eval(Ops.get(@link_update_mapping, ret))
+                if @link_update_mapping.has_key?(ret)
+                  Popup.ShowFeedback(_("Analyzing system"), Message.takes_a_while)
```

Comment:
Use the [new `Popup.Feedback`](https://github.com/yast/yast-yast2/blob/master/library/general/src/modules/Popup.rb#L1078) with block instead (auto closes the popup).